packages/core/contracts/financial-templates/common/FeePayer.sol

```diff
@@ -148,6 +145,27 @@ abstract contract FeePayer is AdministrateeInterface, Testable, Lockable {
         return totalPaid;
     }
 
+    function getOutstandingRegularFees(uint256 time)
+        public
+        view
+        returns (FixedPoint.Unsigned memory regularFee, FixedPoint.Unsigned memory latePenalty)
+    {
+        StoreInterface store = _getStore();
+        FixedPoint.Unsigned memory collateralPool = _pfc();
+
+        // Exit early if there is no collateral from which to pay fees.
+        if (collateralPool.isEqual(0)) {
+            return (regularFee, latePenalty);
+        }
+
+        // Exit early if fees were already paid during this block.
+        if (lastPaymentTime == time) {
+            return (regularFee, latePenalty);
+        }
```

Comment:
nit: I think we end up doing these checks twice in the payRegularFees flow. Could we remove them from that function and only do them here?

Reply:
I had done this to save gas on tx so that the `payRegualrFees` will exit early without calling the `getOutstandingRegularFees` if these null states are true. However, it is code repetition and I'm not actually sure it saves that much as in the worse case when it goes past the null checks it will do the checks twice.

I've refactored this to only check them within the `getOutstandingRegularFees` function.